core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -271,7 +271,9 @@ public RelDataType toType(RelDataTypeFactory typeFactory, Object o) {
 
   public Object toJson(AggregateCall node) {
     final Map<String, Object> map = jsonBuilder.map();
-    map.put("agg", toJson(node.getAggregation()));
+    Map<String, Object> aggMap = new HashMap(toJson(node.getAggregation()));
+    aggMap.put("class", node.getAggregation().getClass().getName());
```

Comment:
Where is this `class` attribute used? It doesn't seem to be in [`RelJsonWriter#toAggCall`](https://github.com/apache/calcite/blob/e7d8d86dd3ab971921f5122de6c95868dec8739c/core/src/main/java/org/apache/calcite/rel/externalize/RelJsonReader.java#L278).

Reply:
Thank you for your comments. This is a problem of lack of class information during udaf serialization. When we deserialize udaf, we will try to load udaf function through class attribute[1].If we can't load SqlAggregateFunction correctly, there will be an exception when creating AggregateCall.
[[1]](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java#L680)

Reply:
why new a map here, is it OK to do like this?
```
final Map<String, Object> aggMap = toJson(node.getAggregation());
aggMap.put("class", node.getAggregation().getClass().getName());
```
And you only need to add class info for UDAF, for system aggregate function, you do not need to add class info in the map.

Reply:
Thanks for review.